Backend/app/provider-platform/dynamic-offer-driver-app/Main/src/Domain/Action/UI/LeaderBoard.hs

```diff
@@ -0,0 +1,63 @@
+{-
+ Copyright 2022-23, Juspay India Pvt Ltd
+
+ This program is free software: you can redistribute it and/or modify it under the terms of the GNU Affero General Public License
+
+ as published by the Free Software Foundation, either version 3 of the License, or (at your option) any later version. This program
+
+ is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without even the implied warranty of MERCHANTABILITY
+
+ or FITNESS FOR A PARTICULAR PURPOSE. See the GNU Affero General Public License for more details. You should have received a copy of
+
+ the GNU Affero General Public License along with this program. If not, see <https://www.gnu.org/licenses/>.
+-}
+
+module Domain.Action.UI.LeaderBoard where
+
+import Data.Aeson
+import Domain.Types.Person
+import Kernel.Prelude
+import qualified Kernel.Storage.Esqueleto as Esq
+import qualified Kernel.Storage.Hedis as Redis
+import Kernel.Types.Common
+import Kernel.Types.Error
+import Kernel.Types.Id
+import Kernel.Utils.Error
+import Storage.CachedQueries.CacheConfig (CacheFlow)
+import qualified Storage.CachedQueries.Merchant as QM
+import qualified Storage.CachedQueries.Merchant.TransporterConfig as CQTC
+import qualified Storage.Queries.DriverStats as QDriverStats
+import qualified Storage.Queries.Person as QPerson
+
+data DriversInfo = DriversInfo
+  { rank :: Int,
+    name :: Text,
+    totalRides :: Int,
+    totalDistance :: Meters
+  }
+  deriving (Generic, Show, ToJSON, FromJSON, ToSchema)
+
+newtype LeaderBoardRes = LeaderBoardRes
+  { driverList :: [DriversInfo]
+  }
+  deriving (Generic, Show, ToJSON, FromJSON, ToSchema)
+
+getDriverLeaderBoard :: (Esq.EsqDBFlow m r, Esq.EsqDBReplicaFlow m r, EncFlow m r, Redis.HedisFlow m r, CacheFlow m r) => Id Person -> Maybe Integer -> m LeaderBoardRes
+getDriverLeaderBoard personId mbLimit = do
+  person <- Esq.runInReplica $ QPerson.findById personId >>= fromMaybeM (PersonDoesNotExist personId.getId)
+  merchant <- QM.findById person.merchantId >>= fromMaybeM (MerchantNotFound person.merchantId.getId)
```

Comment:
Fetch config directly with fetching `merchant`. Anyways if merchant doesn't exist config will also be not there. So, we don't need to check for merchant separately.